lib/rubocop/cop/rails/environment_comparison.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks that Rails.env is compared using `.production?`-like
+      # methods instead of equality against a string or symbol.
+      #
+      # @example
+      #   # bad
+      #   Rails.env == 'production'
+      #
+      #   # bad, always returns false
+      #   Rails.env == :test
+      #
+      #   # good
+      #   Rails.env.production?
+      class EnvironmentComparison < Cop
+        MSG = 'Favor `Rails.env.%s?` over `Rails.env == %s`.'.freeze
+
+        def_node_matcher :environment_comparison?, <<-PATTERN
+          (send
+            (send (const {nil? cbase} :Rails) :env)
+            :==
+            {str sym}
```

Comment:
We can use `$` to capture a node. e.g. `${str sym}` `environment_comparison?(node) do |env| ...`